src/ImageSharp/Image.FromBytes.cs

```diff
@@ -171,5 +172,125 @@ public static Image<TPixel> Load<TPixel>(Configuration config, byte[] data, IIma
                 return Load<TPixel>(config, memoryStream, decoder);
             }
         }
+
+#if !NETSTANDARD1_1
+
+        /// <summary>
+        /// By reading the header on the provided byte array this calculates the images format.
+        /// </summary>
+        /// <param name="data">The byte array containing encoded image data to read the header from.</param>
+        /// <returns>The format or null if none found.</returns>
+        public static IImageFormat DetectFormat(ReadOnlySpan<byte> data)
+        {
+            return DetectFormat(Configuration.Default, data);
+        }
+
+        /// <summary>
+        /// By reading the header on the provided byte array this calculates the images format.
+        /// </summary>
+        /// <param name="config">The configuration.</param>
+        /// <param name="data">The byte array containing encoded image data to read the header from.</param>
+        /// <returns>The mime type or null if none found.</returns>
+        public static unsafe IImageFormat DetectFormat(Configuration config, ReadOnlySpan<byte> data)
+        {
+            fixed (byte* ptr = &data.GetPinnableReference())
+            {
+                using (var stream = new UnmanagedMemoryStream(ptr, data.Length))
```

Comment:
It's unfortunate to allocate a heap object to wrap the span...

Reply:
Makes sense in the `DetectFormat()` use case, thanks!

When doing a full image decoding (`Image.Load(...)`), that allocation is a drop in the bucket however. Using `UnmanagedMemoryStream` seemed to be the best way for "converting" a span into a `Stream`.